### PR TITLE
action buttons: refactor

### PR DESCRIPTION
**Pull request by NRodriguezcuellar**
Refactored the action button and all dependent components to not use the ActionButton component, instead they all now use a normal SUI button component, to keep the interface clear.

needs  https://github.com/inveniosoftware/react-invenio-forms/pull/138

buttons after changes (should be the same):

Screenshot from 2022-06-09 09-47-58
Screenshot from 2022-06-09 09-47-52
Screenshot from 2022-06-09 09-47-47
Screenshot from 2022-06-09 09-47-42

### CODE REVIEW

File: src/lib/components/Creatibutors/CreatibutorsField.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ class CreatibutorsFieldForm extends Component {
       <DndProvider backend={HTML5Backend}>
         <Form.Field
           required={schema === 'creators'}
-          className={creatibutorsError && 'error'}
+          className={creatibutorsError ? 'error' : ''}
```
Comment (kpsherva) [inline src/lib/components/Creatibutors/CreatibutorsField.js:59]: I remember merging this, rebase needed?

File: src/lib/components/DeleteButton.js [hunk 2]
```diff
@@ -113,19 +113,19 @@ export class DeleteButtonComponent extends Component {
...
+              name="delete"
+              onClick={this.handleDelete}
+              icon={(formik) =>
+                formik.isSubmitting && actionState === DRAFT_DELETE_STARTED ? (
+                  <Icon size="large" loading name="spinner" />
+                ) : (
+                  <Icon name="trash alternate outline" />
+                )
+              }
+              content={capitalizedActionLbl }
```
Comment (kpsherva) [inline src/lib/components/DeleteButton.js:127]: can this function be extracted outside the return clause? you would avoid passing callables to the icon prop

File: src/lib/components/DeleteButton.js [hunk 3]
```diff
@@ -113,19 +113,19 @@ export class DeleteButtonComponent extends Component {
...
-                </>
-              )}
-            </ActionButton>
+            <ActionButton
+              {...color}
+              name="delete"
+              onClick={this.handleDelete}
+              icon={(formik) =>
+                formik.isSubmitting && actionState === DRAFT_DELETE_STARTED ? (
+                  <Icon size="large" loading name="spinner" />
```
Comment (kpsherva) [inline src/lib/components/DeleteButton.js:122]: loading should be a prop of a button not the icon... in this case I think you only need to specify when the button should be loading not to introduce the new loading icon

File: src/lib/components/PreviewButton.js [hunk 4]
```diff
@@ -32,21 +32,17 @@ export class PreviewButtonComponent extends Component {
         name="preview"
         isDisabled={(formik) => formik.isSubmitting}
         onClick={this.handlePreview}
-        icon
+        icon={(formik) =>
+          formik.isSubmitting && actionState === DRAFT_PREVIEW_STARTED ? (
+            <Icon size="large" loading name="spinner" />
```
Comment (kpsherva) [inline src/lib/components/PreviewButton.js:37]: you don't need a loading icon, you can simply set loading prop on the button

File: src/lib/components/PublishButton/SubmitReviewButton.js [hunk 5]
```diff
@@ -63,24 +63,22 @@ class SubmitReviewButtonComponent extends Component {
           name="SubmitReview"
           onClick={this.openConfirmModal}
           positive
-          icon
+          icon={(formik) =>
+            formik.isSubmitting &&
+            actionState === 'DRAFT_SUBMIT_REVIEW_STARTED' ? (
+              <Icon size="large" loading name="spinner" />
```
Comment (kpsherva) [inline src/lib/components/PublishButton/SubmitReviewButton.js:69]: same as above

**Review by Pineirin — APPROVED** [review]

File: src/lib/components/Creatibutors/CreatibutorsModal.js [hunk 6]
```diff
@@ -8,10 +8,9 @@
 
 import React, { Component, createRef } from 'react';
 import PropTypes from 'prop-types';
-import { Form, Grid, Header, Icon, Modal } from 'semantic-ui-react';
+import {Button, Form, Grid, Header, Icon, Modal} from 'semantic-ui-react';
```
Comment (kpsherva) [inline src/lib/components/Creatibutors/CreatibutorsModal.js:11]:
```suggestion
import { Button, Form, Grid, Header, Icon, Modal } from 'semantic-ui-react';
```
formatting missing

**Pull request merged**